modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -441,6 +445,13 @@ def broadcast_axis_partitions(
         else:
             num_splits = NPartitions.get()
         preprocessed_map_func = cls.preprocess_func(apply_func)
+
+        if pass_cols_to_partitions:
+            column_func = lambda df: df.columns  # noqa: E731
+            func = cls.preprocess_func(column_func)
+            partition_cols = [part.apply(func) for part in left[0]]
+            column_refs = [col.list_of_blocks[0] for col in partition_cols]
```

Comment:
Maybe it would be better to handle this parameter at the `modin_frame` level and then just pass the `column_refs` as a `func_args` to the partition manager (yeah, we would still have to introduce a new parameter to `part_mgr.broadcast_apply(...func_args...)` but it would have more general appliance).

If we handle the `pass_cols_to_partitions` at the higher level we can easily dispatch on whether to compute refs or get the values from `._columns_cache` if we have some.

Reply:
@dchigarev I raised the processing of this parameter to the dataframe level. Please review.